src/dmd/globals.d

```diff
@@ -362,6 +362,44 @@ struct Global
         _version = (import("VERSION") ~ '\0').ptr;
         compiler.vendor = "Digital Mars D";
     }
+
+    /**
+    Returns: the version as the number that would be returned for __VERSION__
+    */
+    extern(C++) uint versionNumber()
```

Comment:
This code duplicates code in the lexer https://github.com/dlang/dmd/blob/master/src/dmd/lexer.d#L518

In fact, it's a cut&paste of it. It should take an argument instead of relying on a global, and moved to another module (lexer.d?).

Reply:
As you can see later on in this PR, the code was removed from lexer.d and replaced with a call to this function.  I could move this function to lexer.d and have globals import lexer (instead of the other way around).  Would you prefer that?

Reply:
Yes it could take an argument, but then the value won't be cached.  This current implementation allows the value to be cached in case `__VERSION__` is used multiple times along with outputing the `__VERSION__` json value.

Reply:
I see you are correct. I missed that. I with draw both suggestions.